deepy.py

```diff
@@ -17,23 +17,25 @@
 import os
 
 import deepspeed
-from deepspeed.launcher.runner import main
 
-logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
 
-from megatron.neox_arguments import NeoXArgs
-from megatron.utils import get_wandb_api_key
+def main():
+    logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
 
+    from megatron.neox_arguments import NeoXArgs
```

Comment:
What’s the motivation for moving the import inside the main function?

Reply:
[It is a guard to prevent the code from running immediately upon import by another script.](https://stackoverflow.com/questions/419163/what-does-if-name-main-do)

Reply:
I know what main functions do, I’m asking why this specific import is being moved inside the main function.

Reply:
Oh, sorry, I wasn't paying enough attention to what you highlighted. The imports within `main()` are those that are from GPT-NeoX, the others outside of `main()` are from the standard library and external dependencies. This organization is derived from how it was written before I added the guard.